### PR TITLE
Added xeus-cpp

### PR DESCRIPTION
**Pull request by anutosh491**

### CODE REVIEW

**Comment by anutosh491** [comment]
Converting to draft atm !

File: recipes/recipes_emscripten/xeus-cpp/recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  number: 0
+
+requirements:
+  build:
+    - '{{ compiler("cxx") }}'
+    - cmake
+    - make  # [unix]
+  host:
+    - nlohmann_json
+    - xeus-lite
```
Comment (anutosh491) [inline recipes/recipes_emscripten/xeus-cpp/recipe.yaml:22]: Add upper bound
`  - xeus-lite <2.0`

File: recipes/recipes_emscripten/xeus-cpp/recipe.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    - xeus >=3.0.5,<4.0
+    - xtl >=0.7,<0.8
+    - llvm =16.0.6
```
Comment (anutosh491) [inline recipes/recipes_emscripten/xeus-cpp/recipe.yaml:25]: Use `CppInterOp` instead of `llvm`

Comment (mcbarton) [reply recipes/recipes_emscripten/xeus-cpp/recipe.yaml:25]: As this is for a build of 0.3.0 then changing to CppInterOp would be unnecessary. Will be needed 0.4.0 comes out. Also CppInterOp uses llvm 17, so this change will not work if xeus-cpp 0.3.0 is not compatible with llvm 17.

**Comment by vgvassilev** [comment]
I think we can close this PR now as it was superseded by #937.

**Pull request closed without merging**